frontend/public/components/catalog/catalog-items.jsx

```diff
@@ -324,37 +334,61 @@ export class CatalogTileViewPage extends React.Component {
     const { items } = this.props;
 
     if (byKeyword.active) {
-      return this.filterByKeyword(items);
+      return this.filterByKeyword(byKeyword.value, items);
     }
 
     return items;
   }
 
   clearFilters() {
-    const filters = _.cloneDeep(this.state.filters);
-    filters.byKeyword.active = filters.byType.clusterServiceClass.active = filters.byType.imageStream.active = false;
-    filters.byKeyword.value = '';
+    const params = new URLSearchParams(window.location.search);
+    params.delete(KEYWORD_URL_PARAM);
+    params.delete(TYPE_URL_PARAM);
+    const url = new URL(window.location);
+    const searchParams = `?${params.toString()}${url.hash}`;
+    history.replace(`${url.pathname}${searchParams}`);
+    this.setState(this.getUpdatedStateFromURL(searchParams));
     this.filterByKeywordInput.focus();
-    this.setState({filters});
+  }
+
+  updateURL (filterName, value) {
+    const params = new URLSearchParams(window.location.search);
+    if (value) {
+      params.set(filterName, value);
+    } else {
+      params.delete(filterName);
+    }
+    const url = new URL(window.location);
+    const searchParams = `?${params.toString()}${url.hash}`;
+
+    history.replace(`${url.pathname}${searchParams}`);
+    this.setState(this.getUpdatedStateFromURL(searchParams));
   }
 
   onFilterChange(filterType, id, value) {
-    const filters = _.cloneDeep(this.state.filters);
+    const { filters } = this.state;
+    const { byType } = filters;
+    let byTypeValues = '';
+
     if (filterType === 'byKeyword') {
-      const active = !!value;
-      filters[filterType] = { active, value };
+      this.updateURL(KEYWORD_URL_PARAM, `${value}`);
     } else {
-      filters[filterType][id].active = value;
+      byTypeValues = value ? id : ''; // toggle checkbox just selected
+      _.each(Object.keys(byType), typeKey => {
+        if (typeKey !== id) { // add other active byType filters
+          byTypeValues += byType[typeKey].active ? (byTypeValues.length ? `,${typeKey}` : typeKey) : '';
```

Comment:
Make this this an array and use `array.join(',')` or better use JSON in the query param values and `JSON.stringify`